### PR TITLE
Issue #173 Setup CI build on CentOS CI

### LINKED ISSUE #173: Setup CI build on CentOS CI
Using CentOS CI will give us nested virtualization, so that we will be able to run proper integration tests. See also:

* https://wiki.centos.org/QaWiki/CI/Duffy
* https://github.com/projectatomic/vagrant-service-manager/tree/master/.ci

### PR DESCRIPTION
**Pull request by budhrg**
Fix #173

### CODE REVIEW

File: hack/boilerplate/boilerplate.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def file_passes(filename, refs, regexs):
 def file_extension(filename):
     return os.path.splitext(filename)[1].split(".")[-1].lower()
 
-skipped_dirs = ['Godeps', 'third_party', '_gopath', '_output', '.git', 'cluster/env.sh', "vendor", "test/e2e/generated/bindata.go"]
+skipped_dirs = ['Godeps', 'third_party', '_gopath', '_output', '.git', 'cluster/env.sh', "vendor", "test/e2e/generated/bindata.go", "centos_ci.sh"]
```
Comment (hferentschik) [inline hack/boilerplate/boilerplate.py]: Why don't we just add the required boilerplate code?

File: centos_ci.sh [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+mkdir $HOME/gopath $HOME/gopath/src $HOME/gopath/bin $HOME/gopath/pkg
+export GOPATH=$HOME/gopath
+export PATH=$GOROOT/bin:$GOPATH/bin:$PATH
+
+echo $GOPATH
+echo $GOROOT
+
+# Setup minishift repo
+mkdir -p $GOPATH/src/github.com/minishift/minishift
+cp -r `ls` $GOPATH/src/github.com/minishift/minishift/
```
Comment (hferentschik) [inline centos_ci.sh]: instead of copying a symbolic link should do

File: centos_ci.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+cd $GOPATH/src/github.com/minishift/minishift/
+
+# Glide setup
+GLIDE_OS_ARCH=`go env GOHOSTOS`-`go env GOHOSTARCH`
```
Comment (hferentschik) [inline centos_ci.sh]: Instead of all this stuff below have you tried: "curl https://glide.sh/get | sh"

Comment (budhrg) [inline centos_ci.sh]: @hferentschik  Looks like Github API rate limit hit here too https://github.com/Masterminds/glide/issues/708

**Comment by budhrg** [comment]
centos ci is passing here, with travis ci failed due to openshift oc binary github api rate limit.

cc @hferentschik

File: centos_ci.sh [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+GLIDE_TAG=$(curl -s https://glide.sh/version)
+GLIDE_LATEST_RELEASE_URL="https://github.com/Masterminds/glide/releases/download/${GLIDE_TAG}/glide-${GLIDE_TAG}-${GLIDE_OS_ARCH}.tar.gz"
+curl -LO ${GLIDE_LATEST_RELEASE_URL}
+mkdir /tmp/glide
+tar --directory=/tmp/glide -xvf glide-${GLIDE_TAG}-${GLIDE_OS_ARCH}.tar.gz
+export PATH=$PATH:/tmp/glide/${GLIDE_OS_ARCH}
+
+# Test
+make test cross fmtcheck prerelease
```
Comment (hferentschik) [inline centos_ci.sh]: Looking good. I was wondering whether you could look into one more thing though :-) The value proposition of using CentOS CI is to be able to run integration tests (under the assumption that we have nested virtualization on CentOS CI). To confirm this we could run the current integration tests as well.

To make this work you would need to add the _integration_ task here. But you also need to either provision VirtualBox (something I would like to investigate as well) or probably easier make a change to _flags.go_ to read the driver from say an environment variable and if not set defaults to _virtualbox_.

**Pull request merged**